model/convert/service_event_test.go

```diff
@@ -122,6 +122,21 @@ func TestEventConversion(t *testing.T) {
 			assert.Equal(t, expected, actual)
 		},
 	)
+
+	t.Run("protocol state version upgrade", func(t *testing.T) {
+		fixture, expected := unittest.ProtocolStateVersionUpgradeFixtureByChainID(chainID)
+
+		// convert Cadence types to Go types
+		event, err := convert.ServiceEvent(chainID, fixture)
+		require.NoError(t, err)
+		require.NotNil(t, event)
+
+		// cast event type to version upgrade
+		actual, ok := event.Event.(*flow.ProtocolStateVersionUpgrade)
+		require.True(t, ok)
+
```

Comment:
I think we also need to check if fields are correctly converted.

Reply:
That's what the equality assertion on line 138 is doing. Did you have something else in mind?

Reply:
sorry, missed that, all good